api/src/main/java/com/epam/pipeline/manager/cluster/autoscale/ScaleDownHandler.java

```diff
@@ -215,27 +214,16 @@ private void log(final PipelineRun run, final String text) {
     private boolean isNodeAliveOnCloud(final Node node) {
         final String instanceId = getNodeCloudInstanceId(node);
         final String label = getNodeLabel(node);
-        final Optional<InstanceTerminationState> state;
-        final boolean existsOnCloud;
+        final CloudInstanceState state;
         if (isPooled(node)) {
             final NodeRegionLabels cloudRegion = kubernetesManager.getNodeRegion(label);
-            state = cloudFacade.getInstanceTerminationState(cloudRegion, instanceId);
-            existsOnCloud = cloudFacade.instanceExists(cloudRegion, instanceId);
+            state = cloudFacade.getInstanceState(cloudRegion, instanceId);
         } else {
             final Long runId = Long.parseLong(label);
-            final Long cloudRegionId = pipelineRunManager.findRun(runId)
-                    .map(PipelineRun::getInstance)
-                    .map(RunInstance::getCloudRegionId)
-                    .orElseThrow(
-                            () -> new IllegalArgumentException(
-                                    String.format("Can't find region by runId: %d", runId))
-                    );
-            state = cloudFacade.getInstanceTerminationState(cloudRegionId, instanceId);
-            existsOnCloud = cloudFacade.instanceExists(cloudRegionId, instanceId);
+            state = cloudFacade.getInstanceState(runId);
         }
-        log.debug("Check if unavailable node {} is alive. Exists: {}. Termination state: {}", instanceId, existsOnCloud,
-                state.map(InstanceTerminationState::getStateMessage).orElse(EMPTY_TERMINATION_STATE));
-        return existsOnCloud && !state.isPresent();
+        log.debug("Check if unavailable node {} is alive. Termination state: {}", instanceId, state);
+        return state != null && state != CloudInstanceState.TERMINATED;
```

Comment:
When an instance gets preempted on GCP it is getting stopped rather than terminated. So it seems like this won't work for GCP at least. I think that we should check if instance is running.

```suggestion
        return state == CloudInstanceState.RUNNING;
```